packages/arb-util/core/execution.go

```diff
@@ -86,16 +95,23 @@ func (e *ExecutionTracker) fillInCursors(max int) error {
 	return nil
 }
 
-func (e *ExecutionTracker) GetExecutionState(gasUsed *big.Int) (*ExecutionState, *big.Int, error) {
+func (e *ExecutionTracker) GetExecutionCursor(gasUsed *big.Int) (ExecutionCursor, error) {
 	index, ok := e.stopPointIndex[string(gasUsed.Bytes())]
 	if !ok {
-		return nil, nil, errors.New("invalid gas used")
+		return nil, errors.New("invalid gas used")
 	}
 	if err := e.fillInCursors(index); err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
-	cursor := e.cursors[index]
+	return e.cursors[index], nil
+}
```

Comment:
Without a clone this is pretty unsafe since it's a pointer to a cursor that's used internally